OpenRA.Mods.Common/Traits/GivesBounty.cs

```diff
@@ -41,23 +41,48 @@ class GivesBountyInfo : ITraitInfo
 	class GivesBounty : INotifyKilled
 	{
 		readonly GivesBountyInfo info;
+		readonly GainsExperience gainsExp;
+		readonly Cargo cargo;
 
 		public GivesBounty(Actor self, GivesBountyInfo info)
 		{
 			this.info = info;
+			gainsExp = self.TraitOrDefault<GainsExperience>();
```

Comment:
These need to move into `INotifyCreated.Created`, otherwise you introduce yaml-order initialization bugs.